filament/backend/include/backend/Program.h

```diff
@@ -61,7 +63,7 @@ class Program {
 
     using DescriptorBindingsInfo = utils::FixedCapacityVector<Descriptor>;
     using DescriptorSetInfo = std::array<DescriptorBindingsInfo, MAX_DESCRIPTOR_SET_COUNT>;
-    using SpecializationConstantsInfo = utils::FixedCapacityVector<SpecializationConstant>;
+    using SpecializationConstantsInfo = utils::Slice<const SpecializationConstant>;
```

Comment:
so Program is a public API of the backend. I think it's weird that Program must be given an InternPool.

imho, the api/interface to Program should stay an array of spec constant -- but *internally* in the concrete implementations of each backend (or maybe in DriverBase) it can use the internpool

I'm surprised this change doesn't break a bunch of unit tests.


Program is supposed to be a "bundle" that describes the program (mostly shader + spec constants + binding infos). I don't like that it references data outside of itself. It should be a simple value type imho.